### PR TITLE
feat: route to increment the listenings value of a music

### PR DESCRIPTION
**Pull request by TharickABDUL**
## What & How
* New route to increments the value of listenings, pinata metadata

## Tickets
* Ticket : [CU-#3du2nhv](https://app.clickup.com/t/3du2nhv)

### CODE REVIEW

**Comment by Wbebey** [comment]
Overall, this Pull Request looks well organized and clean. Here are a few suggestions for improvement:

In the commit message, it would be more descriptive to specify the route name (e.g., '/api/music/increment-listenings') in addition to the general purpose of the route.

In the MusicController class, it would be a good idea to add some validation for the req.body parameters, specifically musicUrl and listeningsValue. For example, you could add a check to ensure that musicUrl is a valid string and that listeningsValue is a positive integer. This will help prevent any potential errors or unexpected behavior in the application.

In the updateListeningsMetadata method of the PinataService class, it would be helpful to include some error handling in case the update operation fails. This could include logging the error and returning a relevant error message to the client.

It would be a good idea to add some unit tests for the new route and the associated functions in the MusicController and PinataService classes. This will help ensure that the code is working as intended and will make it easier to maintain and make changes in the future.

Overall, this Pull Request looks like a solid addition to the codebase. Keep up the good work!

File: backend/api/src/validators/music.validator.ts [hunk 1]
```diff
@@ -37,6 +37,19 @@ class MusicValidator extends AppValidator implements IMusicValidator {
...
+      throw new AppError(
+        `The musicUrl must not end with a '/'`,
+        StatusCodes.BAD_REQUEST
+      )
+    }
+
+    const musicCIDArray = name.split('/')
+    const musicCID = musicCIDArray[musicCIDArray.length - 1]
+    return musicCID
+  }
```
Comment (alexandre-pinon) [inline backend/api/src/validators/music.validator.ts]: non

**Review by alexandre-pinon — DISMISSED** [review]
LGTM

**Review by alexandre-pinon — APPROVED** [review]
tkt tkt get

**Pull request merged**